roles/setup_minio/tasks/install.yml

```diff
@@ -124,51 +124,54 @@
-    status_code: 200
-  register: response
-  retries: 50
-  delay: 5
-  until: response.status == 200
+- name: Validate minio service using NodePort
+  block:
+    - name: Check minio service availability
+      vars:
+        random_worker_ip: "{{ node_info.resources | json_query('[].status.addresses[?type==`InternalIP`].address') | random | first }}"
+        node_port: "{{ service_info.resources[0].spec.ports[0].nodePort }}"
+      uri:
+        url: "http://{{ random_worker_ip }}:{{ node_port }}/minio/health/ready"
+        return_content: no
+        status_code: 200
+      register: response
+      retries: 50
+      delay: 5
+      until: response.status == 200
 
-- name: Create working directory
-  tempfile:
-    state: directory
-    prefix: minio_
-  register: minio_dir
+    - name: Create working directory
+      tempfile:
+        state: directory
+        prefix: minio_
+      register: minio_dir
 
-- name: Set working directory path
-  set_fact:
-    minio_dir: "{{ minio_dir.path }}"
+    - name: Set working directory path
+      set_fact:
+        minio_dir: "{{ minio_dir.path }}"
 
-- name: Get Minio CLI
-  vars:
-    minio_cli: "https://dl.min.io/client/mc/release/linux-amd64/mc"
-  get_url:
-    url: "{{ minio_cli }}"
-    dest: "{{ minio_dir }}/"
-    mode: "0750"
+    - name: Get Minio CLI
+      vars:
+        minio_cli: "https://dl.min.io/client/mc/release/linux-amd64/mc"
+      get_url:
+        url: "{{ minio_cli }}"
+        dest: "{{ minio_dir }}/"
+        mode: "0750"
 
-- name: Create an initial bucket
-  vars:
-    random_worker_ip: "{{ node_info.resources | json_query('[].status.addresses[?type==`InternalIP`].address') | random | first }}"
-    node_port: "{{ service_info.resources[0].spec.ports[0].nodePort }}"
-    instance: local
-  shell: |
-    {{ minio_dir }}/mc alias set {{ instance }} \ 
-    http://{{ random_worker_ip }}:{{ node_port }} \ 
-    {{ sm_access_key_id }} {{ sm_access_key_secret }} --api S3v4;
-    {{ minio_dir }}/mc mb --ignore-existing {{ instance }}/{{ sm_bucket_name }}
-  register: minio_cmd 
-  retries: 18
-  delay: 5
-  until: minio_cmd is not failed
-  notify:
-    - "Delete temp directory"
-...
\ No newline at end of file
+    - name: Create an initial bucket
+      vars:
+        random_worker_ip: "{{ node_info.resources | json_query('[].status.addresses[?type==`InternalIP`].address') | random | first }}"
+        node_port: "{{ service_info.resources[0].spec.ports[0].nodePort }}"
+        instance: local
+      shell: |
+        {{ minio_dir }}/mc alias set {{ instance }} \
+        http://{{ random_worker_ip }}:{{ node_port }} \
+        {{ sm_access_key_id }} {{ sm_access_key_secret }} --api S3v4;
+        {{ minio_dir }}/mc mb --ignore-existing {{ instance }}/{{ sm_bucket_name }}
+      register: minio_cmd
+      retries: 18
+      delay: 5
+      until: minio_cmd is not failed
+      notify:
+        - "Delete temp directory"
+  when: sm_service_type == "NodePort"
```

Comment:
I have concerns about this only working with nodePort now, as you mentioned with LoadBalancer we also have IPs Nodes IP addresses. We may need to check for a way to always be able to reach the Minio endpoint from the Jump host. Or maybe use the API to create the bucket from inside the cluster using the API.

Reply:
Yes, we can add LoadBalancer type in the condition too because it provides a node-port, ideally we should use the external IP provided by MetalLB, but in our case is not reachable from the jumphost when it's configured in BGP mode, and might be the same for other users, but then we'll need a way to reach the endpoint when using ClusterIP option, and as you said maybe using the API internally. Is this bucket creation only a external test? If workloads want to consume minio, they somehow use the internal endpoint right? If so we can do something : )

Reply:
I synced with Beto, we'll merge this and improve the part that creates the bucket to consider all service type option in another change, we were thinking to create the bucket using the internal service endpoint.